system/jlib/jstats.cpp

```diff
@@ -1756,25 +1756,14 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
 
     CStatisticCollection * ensureSubScope(const StatsScopeId & search, bool hasChildren)
     {
-        //MORE: Implement hasChildren
-        return resolveSubScope(search, true, false);
-    }
+        //Once the CStatisicCollection is created it should not be replaced - so that returned pointers remain valid.
+        CStatisticCollection * match = children.find(&search);
+        if (match)
+            return match;
```

Comment:
Yes.  I'm not 100% sure why it leaked - I could go back and check  I suspect it was if endScope wasn't called.  E.g. because of an exception, but probably other reasons.